### DISCUSSION TITLE
posts/building-my-moments/

### DISCUSSION THREAD

**Discussion by giscus** [WincerChan/blog-comments#8] [Announcements]
# posts/building-my-moments/

早有搭建一个自己的网络相册的想法，只是之前我一直没有找到合适的方案。最先开始想的是一些正经的方案：比如用公共的图床，但是国内公共的图床其实 90% 都不稳定，号称永久免费没多久就倒闭了，而且管理起来也不是很方便；再比如对象存储吧，各种乱七八糟的计费规则：存储空间要收费、读取写入操作要收费、请求也要收费，于是还是放弃了。

https://blog.itswincer.com/posts/building-my-moments/

**Comment by tjsky** [comment]
如果储存超过10GB的话，可以考虑下用backblaze（B2），储存费用比R2要少三分之一，而且B2到CF的流量也是不要钱的。

**Reply by WincerChan** [reply]
谢谢你的回复，我之前还没听说过 backblaze，研究了下发现存储价格只要 0.01 美元！等后续满了是可以考虑迁移到backblaze

**Comment by sdvcrx** [comment]
非常好的想法，不过我发现相册点左边切换照片似乎会加载图片两次

**Reply by WincerChan** [reply]
你说的是切换图片会造成发起两次请求原图是吗？这个确实是一个小问题，原因是我用 Promise 来确保图片被加载完成后，直接返回了 URL 而不是图片的对象，确实不是很优雅，待会我会修复这个问题。

**Reply by WincerChan** [reply]
这个问题应该已经被修复了

**Comment by David222ddd** [comment]
一个很好的方法，有机会我也试一下~，用对象储存一旦存储过多的话费用水涨船高的厉害
（另外为什么只允许github登录捏，这样感觉会少好多评论）

**Reply by WincerChan** [reply]
对，如果一直用原图储存的话，存储空间的费用会涨得很厉害，所以就只能用高压缩的格式来缓解这个问题了。
（其实之前用 Disqus 的时候评论也很少……五年来也就100多条，其中一半还是我的回复。使用目前的 giscus 主要还是考虑到我的博客文章还是偏向技术向，玩技术的人应该都有 GitHub 帐号了，再一个就是 giscus 的评论组件相较于 Disqus 更简洁一些

**Comment by yzyyz1387** [comment]
真的好，我老早前就一直想整一个相册，苦于没能力😂😂，专业也不对口，没有那么多时间精力学技术，羡慕了
另外，2025新年快乐

**Reply by WincerChan** [reply]
谢谢夸奖😋，其实做起来比我想象要简单，另外其实现在市面上有很多产品就提供这种公开式相册，比如：immich shared album，也可以尝试一下下。
也祝你新年快乐🎆
